src/Microsoft.VisualStudio.ProjectSystem.CSharp/ProjectSystem/LanguageServices/CSharpParseCommandLineArguments.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace Microsoft.VisualStudio.ProjectSystem.LanguageServices
+{
+    public class CSharpParseCommandLineArguments : IParseCommandLineArguments
+    {
+        [ImportingConstructor]
+        public CSharpParseCommandLineArguments()
+        {
+        }
+
+        public CommandLineArguments Parse(IEnumerable<string> args, string baseDirectory)
+        {
+            return CommandLineArguments.FromCommonCommandLineArguments(
+                CSharpCommandLineParser.Default.Parse(args, baseDirectory, sdkDirectory: null, additionalReferenceDirectories: null));
+        }
+
+        [Export(typeof(IParseCommandLineArguments))]
```

Comment:
Will do; I'll have to disable the analyzer as per [@srivatsn's comment](https://github.com/dotnet/roslyn-project-system/pull/1876#issuecomment-290239006).